hq/app/schedule/IamAudit.scala

```diff
@@ -1,34 +1,55 @@
 package schedule
 
-import com.gu.anghammarad.models.{AwsAccount => Account}
+import com.gu.anghammarad.models.{Target, AwsAccount => Account}
 import config.Config.{iamAlertCadence, iamHumanUserRotationCadence, iamMachineUserRotationCadence}
 import logic.DateUtils
 import model._
 import org.joda.time.{DateTime, Days}
 import play.api.Logging
-import schedule.IamMessages.createMessage
+import schedule.IamMessages._
 import schedule.IamNotifier.createNotification
 import utils.attempt.FailedAttempt
 
 object IamAudit extends Logging {
 
-  def makeIamNotification(flaggedCreds: Map[AwsAccount, Seq[IAMAlertTargetGroup]]): List[IamNotification] = {
-    flaggedCreds.toList.flatMap { case (awsAccount, targetGroups) =>
+  def makeIamNotification(flaggedCredsToNotify: Map[AwsAccount, Seq[IAMAlertTargetGroup]]): List[IamNotification] = {
+    flaggedCredsToNotify.toList.flatMap { case (awsAccount, targetGroups) =>
       if (targetGroups.isEmpty) {
         logger.info(s"found no IAM user issues for ${awsAccount.name}. No notification required.")
         None
       } else {
         targetGroups.map { tg =>
-          logger.info(s"for ${awsAccount.name}, generating iam notification message for ${tg.users.length} user(s) with outdated keys and missing mfa")
-          tg.users.flatMap { user =>
-            val message = createMessage(tg.users, awsAccount)
-            Some(createNotification(awsAccount, tg.targets :+ Account(awsAccount.accountNumber), message, user.username, createDeadline(user.disableDeadline)))
-          }
+          logger.info(s"for ${awsAccount.name}, generating iam notification message for ${tg.users.length} user(s) with outdated keys and and/or missing mfa")
+          val (usersToReceiveWarningAlerts, usersToReceiveFinalAlerts) = sortAlertsIntoWarningOrFinalAlert(tg.users)
+          createNotifications(warning = true, usersToReceiveWarningAlerts, awsAccount, tg.targets) ++ createNotifications(warning = false, usersToReceiveFinalAlerts, awsAccount, tg.targets)
         }
       }
     }.flatten
   }
 
+  private def sortAlertsIntoWarningOrFinalAlert(users: Seq[VulnerableUser]): (Seq[VulnerableUser], Seq[VulnerableUser]) = {
```

Comment:
I struggled to read this method a bit. This is creating a List where isWarningAlert is true and another where isFinalAlert is true correct? Could the implementation be streamlined a bit by calling `filter` twice instead of a single call to `foldRight`?

Reply:
Thank you so much for this feedback @jorgeazevedo. I love writing `fold` functions 😂  , but I love even more to write readable code, so this feedback is much appreciated and I have changed the function to use `filter` 💖